src/Strategy/RPM.hs

```diff
@@ -0,0 +1,140 @@
+module Strategy.RPM
+  ( discover,
+    getSpecDeps,
+    getTypeFromLine,
+    RPMDependency (..),
+    RequiresType (..),
+    Dependencies (..),
+  )
+where
+
+import Control.Effect.Error
+import qualified Data.Map.Strict as M
+import Data.Maybe (mapMaybe)
+import qualified Data.Text as T
+import DepTypes
+import Discovery.Walk
+import Effect.Grapher
+import Effect.ReadFS
+import Graphing (Graphing)
+import Prologue
+import Types
+
+newtype SpecFileLabel
+  = RequiresType DepEnvironment
+  deriving (Eq, Ord, Show, Generic)
+
+data RPMDependency
+  = RPMDependency
+      { rpmDepName :: Text,
+        rpmConstraint :: Maybe VerConstraint
+      }
+  deriving (Eq, Ord, Show, Generic)
+
+data RequiresType
+  = BuildRequires RPMDependency
+  | RuntimeRequires RPMDependency
+  deriving (Eq, Ord, Show, Generic)
+
+data Dependencies
+  = Dependencies
+      { depBuildRequires :: [RPMDependency],
+        depRuntimeRequires :: [RPMDependency]
+      }
+  deriving (Eq, Ord, Show, Generic)
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \dir _ files ->
+  case find (\f -> ".spec" `isSuffixOf` fileName f) files of
```

Comment:
Can there be more than one `.spec` file in a directory? if so, we'll want to make this a `filter` instead (and use something like `traverse` or `sequence` below to run the strategy on all of the files found)